### DISCUSSION TITLE
Unable to get point co-ordinates

### DISCUSSION THREAD

**Discussion by kmurari103** [iTwin/itwinjs-core#3398] [Q&A] [answered]
Hi there,
I'm upgrading to 3.X but faced an issue when trying to get point co-ordinates by starting a tool. Actually the point returned only contains x & y but z co-ordinate always comes to 0.  I'm attaching a [sandbox ](https://www.itwinjs.org/sandboxes/Krishna%20Kashyap/Get%20Location%20Tool) with issue. However same piece of code worked fine on 2.X. Could anyone point out if i'm missing something here?
Thanks

**Comment by bbastings** [comment, marked as answer]
I don't see why this would have worked differently in 2.x given the same code and test model. You haven't enabled auto-locate or AccuSnap in your tool, so by default an un-snapped point will be projected in view Z to the plane defined by the view's ACS (auxiliary coordinate system, represented by red/green/blue ACS triad decoration shown in your view).

**Reply by kmurari103** [reply]
I actually ended up commenting those piece of code on upgrading to 3.X. Now i uncommented and it started working fine. Thanks for your quick response.
